cmd/vochaintest/vochaintest.go

```diff
@@ -265,16 +283,34 @@ func censusImport(host string, signer *ethereum.SignKeys) {
 	log.Infof("Census created and published\nRoot: %x\nURI: %s", root, uri)
 }
 
-func waitUntilNextBlock(mainClient *client.Client) error {
-	h, err := mainClient.GetCurrentBlock()
-	if err != nil {
-		return fmt.Errorf("cannot get current height: %w", err)
+func (mainClient testClient) ensureTxCostEquals(signer *ethereum.SignKeys, txType models.TxType, cost uint64) error {
+	for i := 0; i < retries; i++ {
+		// if cost equals expected, we're done
+		newTxCost, _ := mainClient.GetTransactionCost(models.TxType_SET_ACCOUNT_INFO)
+		if newTxCost == cost {
+			return nil
+		}
+
+		// else, try to set
+		treasurerAccount, err := mainClient.GetTreasurer()
+		if err != nil {
+			return fmt.Errorf("cannot get treasurer: %w", err)
+		}
+
+		// SetTransactionCost can return OK and then the tx be rejected anyway later.
+		// So, ignore errors, we only care about GetTransactionCost in the next loop
+		_ = mainClient.SetTransactionCost(
```

Comment:
The error should be checked. 

You are correct, the transaction might fail afterwards (when DeliverTx() is executed) but there might be a first error detected by CheckTx() in the moment the TX reaches the mempool.